inst/assets/shinyvalidate.js

```diff
@@ -61,43 +61,53 @@ const bindingStrategy = {
 strategies.push(bindingStrategy);
 
 /**
- * This strategy detects .shiny-input-container at or above the el, and uses
- * Bootstrap 3 classes to display validation messages.
+ * This strategy detects a .form-control within the input el, then:
```

Comment:
I was imagining a separate strategy for BS3 vs BS4 for readability, what do you think?

Reply:
I'm not sure which I like better...let's revisit once all the inputs are working

Reply:
After giving it a bit more thought, I do like the current approach (seems having separate strategies would end up in even more duplicated logic; plus, I think it'd might it a bit harder to reason about when a BS3/BS4 strategy kicks in), but if you think it'd be better, then by all means feel free to refactor